front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ProjectsController.groovy

```diff
@@ -122,6 +127,51 @@ public class ProjectsController {
     return projectDAO.create(project.id, project)
   }
 
+  Set<Project> filter(Collection<Project> projects, Map<String, String> attributes) {
+    attributes = attributes.collect { k,v -> [ k.toLowerCase(), v ] }.collectEntries()
+    Set<Project> items = new HashSet<>()
+
+    if (attributes.containsKey("applications")) {
+      List<String> applications = attributes.applications.split(",")
+      items = projects.findAll { project ->
+        project.config.applications?.find { app -> applications.any { app.toLowerCase().contains(it.toLowerCase()) } } ||
+          project.config.clusters?.find { cluster ->
+            cluster.applications?.collect { app -> applications.any { app.toLowerCase().contains(it.toLowerCase()) } }
+          }
+      }
+      attributes.remove("applications")
+    }
+
+    items += projects.findAll { project ->
+      boolean test = false
+      attributes.each { k, v ->
+        if (fuzzyPredicate(project, k, v)) {
+          test = true
+        }
+      }
+      return test
+    }
+
+    items.sort { Project a, Project b ->
+      return SearchUtils.score(b, attributes) - SearchUtils.score(a, attributes)
+    }
+
+  }
+
+  def fuzzyPredicate = { Project project, String key, String value  ->
+    boolean matches = SearchUtils.matchesIgnoreCase(project.properties, key, value)
+    if (!matches) {
+      if (project.config.pipelineConfigs.find { SearchUtils.matchesIgnoreCase(it.properties, key, value) } ) {
```

Comment:
I suspect that looking into `pipelineConfigs` is probably not providing any value